youtube_dl/extractor/vimeo.py

```diff
@@ -453,11 +453,15 @@ def _real_extract(self, url):
         mobj = re.match(self._VALID_URL, url)
         video_id = mobj.group('id')
         orig_url = url
-        if mobj.group('pro') or mobj.group('player'):
+        if mobj.group('player'):
             url = 'https://player.vimeo.com/video/' + video_id
         elif any(p in url for p in ('play_redirect_hls', 'moogaloop.swf')):
             url = 'https://vimeo.com/' + video_id
 
+        if mobj.group('qs'):
+            # Some vimeopro embeds have query strings (`?portfolio_id=NNNNNN`) which are necessary
+            url += mobj.group('qs')
```

Comment:
`compat_urlparse.urlparse` to extract query from original url. `update_url_query` to pack into new URL. Do not pack if original URL is used.